src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a specific type of Task with an additional deadline(ddl) field.
+ */
+public class Deadline extends Task {
+    private LocalDate ddl;
+
+    /**
+     * The constructor for a Deadline Object.
+     */
+    public Deadline(String taskTitle, LocalDate ddl) {
+        super(taskTitle);
+        this.ddl = ddl;
+    }
+
+    public LocalDate getDdl() {
+        return ddl;
+    }
+
+    /**
+     * Customizes the string representation of a deadline object.
+     *
+     * @return string representation of a deadline in the form [D][{X}] {description} (by: {ddl})
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + "(by: " + ddl.getMonth().toString() + " "
+                + ddl.getDayOfMonth() + " " + ddl.getYear() + ")";
+    }
+
+
+    /**
+     * Two deadline objects are equal iff they are equal tasks and they have the same ddl.
+     * */
+    @Override
+    public boolean equals(Object another) {
+        if (another instanceof Deadline) {
+            Deadline anotherTask = (Deadline) another;
+            return super.equals(another) && ddl.equals(anotherTask.getDdl());
+        }
+        return false;
+    }
```

Comment:
Was this equals method used anywhere?

Reply:
Yes, it is used in the JUnit test ParserTasklistTest to compare whether the expected and actual list of tasks are equal.